homeassistant/components/frigidaire/climate.py

```diff
@@ -0,0 +1,270 @@
+"""ClimateEntity for frigidaire integration."""
+from __future__ import annotations
+
+import logging
+
+import frigidaire
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up frigidaire from a config entry."""
+    client = hass.data[DOMAIN][entry.entry_id]
+
+    def get_entities(username: str, password: str) -> frigidaire.Frigidaire:
+        return client.get_appliances()
+
+    appliances = await hass.async_add_executor_job(
+        get_entities, entry.data["username"], entry.data["password"]
+    )
+
+    async_add_entities(
+        [FrigidaireClimate(client, appliance) for appliance in appliances],
+        update_before_add=True,
+    )
+
+
+class FrigidaireClimate(ClimateEntity):
+    """Representation of a Frigidaire appliance."""
+
+    def __init__(self, client, appliance):
+        """Build FrigidaireClimate.
+
+        client: the client used to contact the frigidaire API
+        appliance: the basic information about the frigidaire appliance, used to contact the API
+        """
+
+        self._client = client
+        self._appliance = appliance
+        self._details = None
+
+        # Entity Class Attributes
+        self._attr_unique_id = appliance.appliance_id
+        self._attr_name = appliance.nickname
+        self._attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+        self._attr_target_temperature_step = 1
+
+        # Although we can access the Frigidaire API to get updates, they are
+        # not reflected immediately after making a request. To improve the UX
+        # around this, we set assume_state to True
+        self._attr_assumed_state = True
```

Comment:
Frigidaire is [eventually consistent](https://en.wikipedia.org/wiki/Eventual_consistency) which requires this hack for things to feel right